test/integration/targets/cloudscale_server/tasks/main.yml

```diff
@@ -39,6 +39,92 @@
       - server is not changed
       - server.state == 'running'
 
+- name: Test update a running server without force in check mode
+  cloudscale_server:
+    name: '{{ cloudscale_resource_prefix }}-test'
+    flavor: '{{ cloudscale_test_flavor_2 }}'
+    image: '{{ cloudscale_test_image }}'
+    ssh_keys: '{{ cloudscale_test_ssh_key }}'
+    force: no
+  register: server
+  check_mode: yes
+- name: Verify update a running server without force in check mode
+  assert:
+    that:
+      - server is not changed
+      - server.state == 'running'
+
+- name: Test update a running server without force
+  cloudscale_server:
+    name: '{{ cloudscale_resource_prefix }}-test'
+    flavor: '{{ cloudscale_test_flavor_2 }}'
+    image: '{{ cloudscale_test_image }}'
+    ssh_keys: '{{ cloudscale_test_ssh_key }}'
+    force: no
+  register: server
+- name: Verify update a running server without force
+  assert:
+    that:
+      - server is not changed
+      - server.state == 'running'
+
+- name: Test update a running server without force idempotence
+  cloudscale_server:
+    name: '{{ cloudscale_resource_prefix }}-test'
+    flavor: '{{ cloudscale_test_flavor_2 }}'
+    image: '{{ cloudscale_test_image }}'
+    ssh_keys: '{{ cloudscale_test_ssh_key }}'
+    force: no
+  register: server
+- name: Verify update a running server without force idempotence
+  assert:
+    that:
+      - server is not changed
+      - server.state == 'running'
+
+- name: Test update a running server with force in check mode
+  cloudscale_server:
+    name: '{{ cloudscale_resource_prefix }}-test'
+    flavor: '{{ cloudscale_test_flavor_2 }}'
+    image: '{{ cloudscale_test_image }}'
+    ssh_keys: '{{ cloudscale_test_ssh_key }}'
+    force: yes
+  register: server
+  check_mode: yes
+- name: Verify update a running server with force in check mode
+  assert:
+    that:
+      - server is changed
+      - server.state == 'running'
```

Comment:
This should also check `server.flavor == cloudscale_test_flavor` to ensure the flavor is not changed in check mode.

This also applies to the other tests above if we keep the current behavior of the force parameter. But see my other comment on that.